jetcd-core/src/main/java/io/etcd/jetcd/resolver/IPNameResolver.java

```diff
@@ -61,21 +60,24 @@ public IPNameResolver(URI targetUri) {
         this.lock = new Object();
         this.targetUri = targetUri;
         this.authority = targetUri.getAuthority() != null ? targetUri.getAuthority() : SCHEME;
+        // we need override authority when authority not set.
+        final boolean needOverrideAuthority = targetUri.getAuthority() == null;
         this.addresses = Stream.of(targetUri.getPath().split(","))
             .map(address -> {
                 return address.startsWith("/")
                     ? address.substring(1)
                     : address;
             })
+            .map(URI::create)
             .map(address -> {
-                Iterable<String> split = Splitter.on(':').split(address);
-                String host = Iterables.get(split, 0);
-                String port = Iterables.get(split, 1, ETCD_CLIENT_PORT);
-                return new InetSocketAddress(host, Integer.parseInt(port));
-            }).map(address -> {
                 return new EquivalentAddressGroup(
-                    address,
-                    io.grpc.Attributes.EMPTY);
+                    new InetSocketAddress(address.getHost(), address.getPort()),
+                    needOverrideAuthority
```

Comment:
this can be simplified by changing the way` this.authority` is handled as we can leave it null if the authority needs to be overridden (of course the getAuthority method need to be updated)